### DISCUSSION TITLE
Anything at all lets chat about this and all work as a big team haha

### DISCUSSION THREAD

**Discussion by blkph0x** [blkph0x/CVE_2024_30078_POC_WIFI#8] [General]
hello guys, so as this is still a work in progess and from the looks of things we may be the only ones working on this in the public realm so any help is welcome, ideas, issues try and keep any ideas and questions here and limit the issues page to code issues.

Thanks,
BlkPh0x

**Comment by AntiqueSquid** [comment]
Made any progress with this CVE PoC recently? Been looking into myself with a small group, but not been able to get the exploit to 'pop'.

**Reply by blkph0x** [reply]
yeah.. so it seems the talked this up a lot there are some pre requests for this to be exploited as it currently is. 
As previously noted, the native driver does not account for 4 extra bytes when the ether type is 0x8100 (Vlan tagged net).
Since the packet is rewritten on the original buffer, it will do so 4 bytes ahead of what is was supposed to because of the dot1q header that should exist.
This will not only corrupt the translated packet (which will be later discarded as invalid) but also overwrite up to 2 bytes outside the MDL buffer that holds the packet. [overflowed bytes marked in red]
But is the exploitability feasible?
Well... not easily since there's a check that ensures the 12 bits of the vlan ID field must be zero (at the location where the dot1q header was expected).
In order to overflow the buffer you can't send those extra bytes so you depend on what was there in the first place.
You can either send 1 zero byte after the SNAP header overflowing 1 byte, or none at all and overflow 2 bytes. Keep in mind you must brute force sending packets until the 2 bytes after the buffer are both zero. Possible but not easy since there are 65535 possibilites.
This is not where the hurdle stops. The heap memory disposition adjacent to the buffer matters. You do control the 2 bytes that are overwritten (they'll be the last 2 octets of the transmitter MAC address) but what you overwrite is important and will decide if there's a crash.
If you're lucky, you might overwrite some memory pointer that is later dereferenced and cause a crash but the exploitability to get remote code execution will be very very slim.
At the very least you'd need another vulnerability that remotely leaks a kernel pointer.
So, in a nutshell, the impact of the CVE is imho much less severe than anticipated from the
Microsoft

advisory.
You need to share the same network as the target, need to brute force a specific memory layout and still need extra information to perform anything other than a DoS.

**Reply by blkph0x** [reply]
https://x.com/f4rmpoet/status/1807055596769751091

**Comment by AntiqueSquid** [comment]
Thanks for the update, very useful to see. Was hoping to continue developing a RCE poc for this with my team, but what you have said seems to align with our thoughts - a consistent RCE seems improbable. Still working towards a working DoS PoC ourselves, and using an old Win 10 build as our victim. Might eventually try to chain with other vulnerabilities, but seems to be an exploit destined for educational purposes rather than a real threat. I do agree this CVE is not as severe as Microsoft seems to imply - unless there is a way to create a consistent RCE yet to be released. 

Have you managed a consistent DoS, or is it a bit hit and miss?
